core/src/components/searchbar/searchbar.scss

```diff
@@ -64,3 +64,14 @@ ion-searchbar {
 .searchbar-has-value.searchbar-has-focus .searchbar-clear-icon {
   display: block;
 }
+
+// adding rtl support 
+[dir="rtl"]  .searchbar-search-icon {
+  left: auto;
+  right: 16px;
+}
```

Comment:
I am unable to run this for the next 3 days, any chance for a before & after pic?

Can you also explain why `16px` exactly? We would want to make it a variable in case it is adaptive.

Reply:
just a convention, for users to be able to control their bundles and directions, we do:
```scss
 .searchbar-search-icon {
    @include rtl() {
      left: auto;
      right: 16px;
     }
  }
```

Reply:
@AmitMY can you explain please?

Reply:
I'm not sure to what you are referring, so I'll answer both.

- If `16px` has some meaning, we would prefer it to be `$searchbar-search-icon-start: 16px`
- Writing `[dir="rtl"]` while nice, is forceful. Ionic allows the user to select their language as `ltr`, and when doing so having `[dir="rtl"]` in the code is just bloat. Instead, we use `@include rtl()` which is a mexin we wrote that adds `[dir="rtl"]` only for users who want their application to work in right-to-left, and for ltr users doesn't do anything.